File: tile.go
/* */

package main

type tile struct {
	Terrain  string `json:"terrain"`
	Blocked  bool   `json:"blocked"`
	BlockLOS bool   `json:"blockLOS"`
	Explored bool   `json:"explored"`
	Visible  bool   `json:"visible"`
}

func (t tile) set(terrain string) tile {
	switch terrain {
	case "floor":
		return tile{terrain, false, false, false, false}
	case "wall":
		return tile{terrain, true, true, false, false}
	case "unknown":
		return tile{terrain, true, true, false, false}
	}
	return tile{}
}

File: __prologue/core/map/tile.js
/* */

console.log('Loading..... map/tile.js');

class Tile {
  constructor(terrain) {
    this.createTile(terrain);
  }
  createTile(terrain) {
    let p = [];
    switch (terrain) {
      case "floor":
        p = [terrain, true, false, false, false];
        break;
      case "wall":
        p = [terrain, false, true, false, false];
        break;
      case "unknown":
        p = [terrain, false, true, false, false];
        break;
      default:
        p = [];
    }
    this.terrain = terrain;
    this.walkable = p[1];
    this.blockLOS = p[2];
    this.explored = p[3];
    this.visible = p[4];
  }
}

export {
  Tile
};
